impeller/geometry/path_component.cc

```diff
@@ -6,15 +6,84 @@
 
 #include <cmath>
 
+#include "impeller/geometry/scalar.h"
 #include "impeller/geometry/wangs_formula.h"
 
 namespace impeller {
 
-VertexWriter::VertexWriter(std::vector<Point>& points,
-                           std::vector<uint16_t>& indices)
+/////////// FanVertexWriter ///////////
+
+FanVertexWriter::FanVertexWriter(Point* point_buffer, uint16_t* index_buffer)
+    : point_buffer_(point_buffer), index_buffer_(index_buffer) {}
+
+FanVertexWriter::~FanVertexWriter() = default;
+
+size_t FanVertexWriter::GetIndexCount() const {
+  return index_count_;
+}
+
+void FanVertexWriter::EndContour() {
+  if (count_ == 0) {
+    return;
+  }
+  index_buffer_[index_count_++] = 0xFFFF;
```

Comment:
Probably being paranoid, but can you double check that the restart markers are the same in all APIs?

Reply:
Nevermind, I checked that they are.

Reply:
I think in GL you can configure a specific restart value with some APIs but I hope to never need that

Reply:
Isn't that desktop GL? On ES 3.0 and above, you specify `GL_PRIMITIVE_RESTART_FIXED_INDEX` to `glEnable` and get the same behavior as Metal and Vulkan.

Reply:
You can't configure the index in ES.